bin/cli.js

```diff
@@ -19,6 +19,7 @@ program
     '--public-url <url>',
     'set the public URL to serve on. defaults to the same as the --out-dir option'
   )
+  .option('--https', 'serves files over HTTPS')
```

Comment:
Rather than making this an option, could we just start the server on both http and https so they both work at the same time?

Reply:
though i suppose they'd need to be on different ports :/

Reply:
I would imagine that generating TLs Crete is fairly expensive, which is a problem when we’re going for efficiency.

Reply:
**TLS certs

Reply:
If that's an actual problem we could always cache them.

Reply:
Looking at `src/utils/generateCertificate.js`, it already caches the generated cert and serves it from the fs automatically. Paying the price *once* on first-run might not be such a big deal. It does raise these questions, though:

- Which does `--open` launch?
- Do we log both `address:port` combinations?
- Do we have a `--no-https` to disable it?

Reply:
Yes I do cache the generated Certificate. We can have both servers running and redirect to https, or have only https running all the time and only disable it by adding `--no-https` option.

Reply:
Would there ever be a reason to _not_ serve over HTTPS? I say we default to HTTPS by default and add a `--no-https` to disable it.

Reply:
> Would there ever be a reason to not serve over HTTPS?

It's a self-signed cert, which the browser is going to warn you about with a big scary ⚠️ message. I think making https opt-in is ideal. Use it when the use case warrants it.